server/src/main/java/io/pravega/schemaregistry/storage/impl/schemas/SchemaRecords.java

```diff
@@ -421,6 +423,10 @@ private void read00(RevisionDataInput source, SchemaGroupsList.SchemaGroupsListB
     @SneakyThrows(IOException.class)
     @SuppressWarnings("unchecked")
     static <T extends Value> T fromBytes(Class<? extends Key> keyClass, byte[] bytes, Class<T> valueClass) {
+        val versionSerializer = SERIALIZERS_BY_KEY_TYPE.get(keyClass);
+        if ( versionSerializer == null ) {
+            throw new SerializationException(String.format("No serializer found for the class %s", keyClass.toGenericString()));
+        }
         return (T) SERIALIZERS_BY_KEY_TYPE.get(keyClass).deserialize(bytes);
```

Comment:
@a6dulaleem Can we use `versionSerializer` at line 430

Reply:
sure